### PR TITLE
Target .NET 10 / net10.0

### PR DESCRIPTION
**Pull request by ViktorHofer**
Contributes to https://github.com/dotnet/runtime/issues/105343

### CODE REVIEW

File: src/libraries/System.Diagnostics.DiagnosticSource/src/System.Diagnostics.DiagnosticSource.csproj
Comment (akoeplinger) [inline src/libraries/System.Diagnostics.DiagnosticSource/src/System.Diagnostics.DiagnosticSource.csproj:136]: also use TFI here?

File: eng/resolveContract.targets
Comment (akoeplinger) [inline eng/resolveContract.targets:61]: hah, I can't wait to discover all the other fun cases like these with `net10.0` 😄

File: src/libraries/System.Diagnostics.DiagnosticSource/src/System.Diagnostics.DiagnosticSource.csproj
Comment (ViktorHofer) [reply src/libraries/System.Diagnostics.DiagnosticSource/src/System.Diagnostics.DiagnosticSource.csproj:136]: This library also targets net8.0 so we need to keep this condition.

File: eng/resolveContract.targets
Comment (ViktorHofer) [reply eng/resolveContract.targets:61]: You were spot on, see my next commit :D

**Comment by carlossanlop** [comment]
@ViktorHofer I already had this PR open for this purpose: https://github.com/dotnet/runtime/pull/106421

**Comment by carlossanlop** [comment]
I closed mine in favor of yours.

**Comment by ViktorHofer** [comment]
> @ViktorHofer I already had this PR open for this purpose: https://github.com/dotnet/runtime/pull/106421

@carlossanlop this PR handles the re-targeting part, not the re-branding (as documented in https://github.com/dotnet/runtime/issues/105343). We need your re-branding PR in the sooner the better.

**Comment by jtschuster** [comment]
It looks like bumping the SDK version and not the NativeAOT runtime package version is causing failures related to https://github.com/dotnet/runtime/pull/106360 in NativeAOT compilations (csc compiles against the runtime libs in the global.json sdk, but ilc compiles against the runtime libs in the `runtime.win-x64.microsoft.dotnet.ilcompiler` package which is from before `MemoryMarshal` type was moved). Bumping the `MicrosoftDotNetILCompilerVersion` to match the SDK should fix the failures.

**Comment by ViktorHofer** [comment]
@StephenMolloy can you please take a look at the System.Runtime.Serialization.Schema test failures?

https://helixre8s23ayyeko0k025g8.blob.core.windows.net/dotnet-runtime-refs-pull-106599-merge-7c3939ad93ba46908e/System.Runtime.Serialization.Schema.Tests/1/console.7f2d10c7.log?helixlogtype=result

The tests are probably not resilient against the assembly version change:

> [System.CodeDom.Compiler.GeneratedCodeAttribute("System.Runtime.Serialization.Schema", "10.0.0.0")]

> Expected: 1565
Actual:   1566

**Comment by akoeplinger** [comment]
I looked at the System.Runtime.Serialization.Schema test failures and yeah they check for a certain character count and now we have one more with 10.0.

Given we'll hit this the next time with .NET 100 I just bumped the count rather than doing something more complicated.

**Comment by lewing** [comment]
@ilonatommy can you take a look at the signalr wbt, here we may need to disable them temporarily

**Comment by ilonatommy** [comment]
> @ilonatommy can you take a look at the signalr wbt, here we may need to disable them temporarily

Yes, we have to disable, we cannot upgrade them to target net10 yet because sdk for net10 is not available yet:
`The current .NET SDK does not support targeting .NET 10.0.  Either target .NET 9.0 or lower, or use a version of the .NET SDK that supports .NET 10.0.`

File: src/libraries/System.Runtime.Serialization.Schema/tests/System/Runtime/Serialization/Schema/Import/ImporterTests.cs
Comment (StephenMolloy) [inline src/libraries/System.Runtime.Serialization.Schema/tests/System/Runtime/Serialization/Schema/Import/ImporterTests.cs:115]: FYI, this file has been updated in preparation for future serializer development. (#104173) The same approach to fixing applies (same 3 updates here actually) but there are a few additional lines to update the character count on.

Comment (ViktorHofer) [reply src/libraries/System.Runtime.Serialization.Schema/tests/System/Runtime/Serialization/Schema/Import/ImporterTests.cs]: @StephenMolloy can you please take care of that?

**Comment by ViktorHofer** [comment]
I will be out for the next six weeks. Anyone feel free to take this over and complete the re-targeting.

**Comment by ericstj** [comment]
@ViktorHofer is it just to drive this to green, or are there other things we need to bring into here to make it complete?

**Comment by ViktorHofer** [comment]
Drive to green. What remains is fixing a few misbehaving tests and possibly browser/wasm and workloads stuff which tends to break with these major version bumps. It would also be good to communicate this change to other teams so that they aren't surprised when we start targeting net10.0.

File: src/libraries/System.Runtime.Serialization.Schema/tests/System/Runtime/Serialization/Schema/Import/ImporterTests.cs
Comment (StephenMolloy) [reply src/libraries/System.Runtime.Serialization.Schema/tests/System/Runtime/Serialization/Schema/Import/ImporterTests.cs]: Done

File: src/libraries/System.Resources.Extensions/tests/TestData.resources
Comment (am11) [inline src/libraries/System.Resources.Extensions/tests/TestData.resources:1]: Maybe we should re-index this as a text file so we can see the diffs in web view?

```sh
echo "*.resources text" >> .gitattributes
git rm --cached src/libraries/System.Resources.Extensions/tests/TestData.resources
git add .gitattributes src/libraries/System.Resources.Extensions/tests/TestData.resources
git commit -m "Treat .resources files as text"
```

**Comment by lewing** [comment]
I'm not sure what is going on with the importer tests, either they are running against multiple targets or it changed again?

**Comment by lewing** [comment]
> Drive to green. What remains is fixing a few misbehaving tests and possibly browser/wasm and workloads stuff which tends to break with these major version bumps. It would also be good to communicate this change to other teams so that they aren't surprised when we start targeting net10.0.

I think the workload stuff is under control at the moment but I will keep an eye on it

**Comment by StephenMolloy** [comment]
> I'm not sure what is going on with the importer tests, either they are running against multiple targets or it changed again?

The test is dynamically generating code which contains attributes that look like this:
```cs
[System.CodeDom.Compiler.GeneratedCodeAttribute("System.Runtime.Serialization.Schema", "9.0.0.0")]
```

This was causing failures early on in this PR journey because that version was updated to '10.0.0.0'. So the importer tests were updated. But looking at the current failures, the generated code is spitting out '9.0.0.0' again. [This is the importer code that adds the attribute to the generated code](https://github.com/ViktorHofer/runtime/blob/bd7a2b99c30b1ab2aebd7993549d080351009e17/src/libraries/System.Runtime.Serialization.Schema/src/System/Runtime/Serialization/Schema/CodeExporter.cs#L520-L522):
```
            AssemblyName assemblyName = Assembly.GetExecutingAssembly().GetName();
            generatedCodeAttribute.Arguments.Add(new CodeAttributeArgument(new CodePrimitiveExpression(assemblyName.Name!)));
            generatedCodeAttribute.Arguments.Add(new CodeAttributeArgument(new CodePrimitiveExpression(assemblyName.Version?.ToString()!)));
```

Somehow, the tests are running with a 9.0 Sys.RT.Serialization.Schema package now, but they were using a 10.0 version before? 🤷‍♂️

**Comment by StephenMolloy** [comment]
For the importer tests, if somebody can figure out why the System.Runtime.Serialization.Schema package has the wrong version, I'd guess that would be the ideal. But if that is potentially an odd artifact of the way our builds work (since the Schema package is just runtime-adjacent 'OOB'), then the tests could be fixed to be more robust against version changes (so we don't run into this again in .Net 100 😆) by cherry picking https://github.com/StephenMolloy/runtime/commit/be7f2ee3d4b750f3085f1c92383dc2a260f505a0

File: src/libraries/System.Resources.Extensions/tests/TestData.resources
Comment (am11) [reply src/libraries/System.Resources.Extensions/tests/TestData.resources:1]: Ah I thought it’s an xml file.

**Comment by ViktorHofer** [comment]
> For the importer tests, if somebody can figure out why the System.Runtime.Serialization.Schema package has the wrong version, I'd guess that would be the ideal.

Fixed with latest commit. That was a bad merge.

File: src/libraries/System.Numerics.Tensors/src/CompatibilitySuppressions.xml
Comment (akoeplinger) [inline src/libraries/System.Numerics.Tensors/src/CompatibilitySuppressions.xml:132]: this is because of https://github.com/dotnet/runtime/pull/107097#pullrequestreview-2267229024
@ericstj I'd have expected APICompat to fail on that PR already, any idea why it only shows up here?

**Comment by akoeplinger** [comment]
@adamsitnik any idea why `System.Runtime.Serialization.Formatters.Tests.BinaryFormatterTests.ValidateAgainstBlobs` only fails on the `windows-x86 Release Libraries_NET48` job?

**Comment by adamsitnik** [comment]
> @adamsitnik any idea why `System.Runtime.Serialization.Formatters.Tests.BinaryFormatterTests.ValidateAgainstBlobs` only fails on the `windows-x86 Release Libraries_NET48` job?

I took a quick look. I was suprised to see `System.Resources.Extensions.BinaryFormat.Tests.csproj` not failing (it uses the same tests), but just realized that it's not executed for Full Framework:

https://github.com/dotnet/runtime/blob/3f103db9ee41062b563cadb91ea23f92e4b978bd/src/libraries/System.Resources.Extensions/tests/BinaryFormatTests/System.Resources.Extensions.BinaryFormat.Tests.csproj#L3

To tell the long story short this test serializes given object with BF, then tries to normalize the produced byte array and compare it against what is stored in the test project (as a base64 string).

I expect that part of the normalization fails. Example: we replace the version number here:

https://github.com/dotnet/runtime/blob/3f103db9ee41062b563cadb91ea23f92e4b978bd/src/libraries/System.Runtime.Serialization.Formatters/tests/BinaryFormatterTests.cs#L677

But I have no idea what it happens only for Full Framework. I can debug it. 

@akoeplinger how important is this? (I need to prioritize my work items)

**Comment by akoeplinger** [comment]
@adamsitnik I debugged this and figured out the issue. It is trying to compare strings which look like this before the normalization:

```
pSystem.Runtime.Serialization.Formatters.Tests, Version=4.0.3.0, Culture=neutral, PublicKeyToken=9d77cc7ad39b68eb
qSystem.Runtime.Serialization.Formatters.Tests, Version=10.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51
```

Notice the `p` and `q` at the start, these are the string lengths (112 and 113 in ASCII) so we fail the comparison since those "chars" are still different even after normalizing the version to 0.0.0.0.

I think the easiest option would be to update the stored blobs and increase the string length by e.g. putting `4.0.30.0` as the version so the lengths match. That should give us breathing room until .NET 100 😄 

Oh and the reason why it only fails on net48 is because the tests aren't running on netcore, even after I merged https://github.com/dotnet/runtime/pull/106858 into the branch.

**Comment by adamsitnik** [comment]
> so we fail the comparison since those "chars" are still different even after normalizing the version to 0.0.0.0

@akoeplinger great find!

> I think the easiest option would be to update the stored blobs and increase the string length

This check predates me, TBH I need to wrap my head around it and figure out whether we benefit anything from having it. BF needs to be able to deserialize inputs serialized with previous .NET versions anyway, so I think the test could just do the following:
- if there is no diff, work with the provided input
- if there is a diff, work with the provided input and with the data generated by BF for current TFM. Just make sure both work.

This would give us the best of both words, without the need to update anything.

But as I said, I need to ensure that I am not missing any important scenario.

cc @JeremyKuhne @stephentoub @bartonjs who may know why this check was added in the first place

File: src/libraries/System.Numerics.Tensors/src/CompatibilitySuppressions.xml
Comment (ericstj) [reply src/libraries/System.Numerics.Tensors/src/CompatibilitySuppressions.xml:132]: It's brand new API so we are allowed to break it.  In 9.0 we're comparing to the 8.0.0 packages.  Here in 10.0 we're comparing to the 9.0 preview7 packages.

**Comment by adamsitnik** [comment]
> Oh and the reason why it only fails on net48 is because the tests aren't running on netcore in the System.Runtime.Serialization.Formatters.Tests project, even after I merged https://github.com/dotnet/runtime/pull/106858 into the branch.

Most of these tests are going to be executed only in Release:

https://github.com/dotnet/runtime/blob/46c9a4fff83f35ec659e6659050440aadccf3201/src/libraries/System.Runtime.Serialization.Formatters/tests/BinaryFormatterTests.cs#L51

On Core they are going to be disabled only when the MsBuild property is not found or set to false:

https://github.com/dotnet/runtime/blob/46c9a4fff83f35ec659e6659050440aadccf3201/src/libraries/System.Runtime.Serialization.Formatters/tests/TestConfiguration.cs#L32-L35

And I thought they are always enabled:

https://github.com/dotnet/runtime/blob/46c9a4fff83f35ec659e6659050440aadccf3201/Directory.Build.props#L475

And then found:

https://github.com/dotnet/runtime/blob/46c9a4fff83f35ec659e6659050440aadccf3201/src/libraries/System.Runtime.Serialization.Formatters/tests/System.Runtime.Serialization.Formatters.Tests.csproj#L68-L78

Since BF needs to throw for `NetCoreAppCurrent`, it seems that we should just enable those tests for `NetCoreAppMinimum` (so we would test both scenarios)

https://github.com/dotnet/runtime/blob/46c9a4fff83f35ec659e6659050440aadccf3201/src/libraries/System.Runtime.Serialization.Formatters/src/System.Runtime.Serialization.Formatters.csproj#L15

@akoeplinger does adding `NetCoreAppMinimum` tfm to the test project has any downsides?

The alternative would be to have two test projects:
- one that targets `NetCoreAppCurrent` and has no custom setting and just ensures that BF throws
- another one that also targets `NetCoreAppCurrent`, but references `NetCoreAppMinimum` of the library project and ensures it works as expected.

**Comment by akoeplinger** [comment]
hmm good question, we only build tests for NetCoreAppCurrent by default:

https://github.com/dotnet/runtime/blob/46c9a4fff83f35ec659e6659050440aadccf3201/src/libraries/tests.proj#L5

@ericstj what do you think?

**Comment by akoeplinger** [comment]
Actually, I don't think we need to have a test project that targets `NetCoreAppMinimum`, we just need to reference the `NetCoreAppMinimum` version of System.Runtime.Serialization.Formatters.csproj (like the other projects do). But I think this will require having two test projects.

**Comment by ericstj** [comment]
We don't run tests against older framework versions today.  There's an issue for this somewhere, but I don't think that's really what we want.  It's not like we care more about folks using the binary formatter compat package on the previous version - that's just a quirk with how the library chose to represent it's OOB version.

I think a separate test project for actual behavior (different from PNSE) that tests the library on the latest framework is a better approach.

**Comment by adamsitnik** [comment]
> I think a separate test project for actual behavior (different from PNSE) that tests the library on the latest framework is a better approach.

I agree, it turned out to be a relatively simple change: https://github.com/dotnet/runtime/pull/107408

**Comment by akoeplinger** [comment]
Talked with Adam and we'll hardcode the test assembly version to 9.0 so we don't need to touch the majority of the tests (only a handful that embed e.g. System.Collections, Version 10.0.0.0)

**Comment by lewing** [comment]
https://github.com/dotnet/sdk/pull/43070 has the sdk first work

**Comment by lewing** [comment]
This looks in pretty good shape now, @ilonatommy can you take a look at why the perf lanes and http functional tests are failing?

**Comment by am11** [comment]
This might fix the perf leg: https://github.com/dotnet/performance/pull/4470. You can quickly try it, to find out if there is more work to be done, by temporarily switching to my branch:

```diff
--- a/eng/pipelines/coreclr/templates/build-perf-bdn-app.yml
+++ b/eng/pipelines/coreclr/templates/build-perf-bdn-app.yml
@@ -70,7 +70,7 @@ steps:
   - script: |
       set -x
       pwd
-      git clone https://github.com/dotnet/performance.git --depth 1 -b ${{parameters.perfBranch}} --single-branch
+      git clone https://github.com/am11/performance.git --depth 1 -b patch-1 --single-branch
     displayName:  Clone performance
     workingDirectory: $(Build.SourcesDirectory)
```

**Comment by ilonatommy** [comment]
> This looks in pretty good shape now, @ilonatommy can you take a look at why the perf lanes and http functional tests are failing?

Http functional tests are not from this PR: https://github.com/dotnet/runtime/issues/108019. Will update about perf when I have more info.

**Comment by am11** [comment]
Pushed another change, could you please rerun (just) that leg?

**Comment by am11** [comment]
Strange that it's neither finding startswith("9.0") nor startswith("10.0") subdir. Added some diagnostics to see which SDKs are available. One more try. :)

**Comment by am11** [comment]
Sorry it was a wrong pipeline, this is the correct one:
```diff
--- a/eng/pipelines/runtime-wasm-perf.yml
+++ b/eng/pipelines/runtime-wasm-perf.yml
@@ -32,9 +32,9 @@ extends:
           runProfile: 'v8'
           collectHelixLogsScript: ${{ variables._wasmCollectHelixLogsScript }}
           onlySanityCheck: true
-          #perfForkToUse: # dummy change
-            #url: https://github.com/radical/performance
-            #branch: fix-build
+          perfForkToUse: # dummy change
+            url: https://github.com/am11/performance
+            branch: patch-1
           #downloadSpecificBuild:
             #buildId: '1878694'
             #pipeline: 'perf-wasm'
```

**Comment by am11** [comment]
Still using dotnet/performance and main branch

image


maybe it is using these pipelines from runtime's `main` branch instead of PR's?

**Comment by ilonatommy** [comment]
Shouldn't we change this line?
https://github.com/dotnet/runtime/blob/89f245e722eda87869dbb04690cb91361af8174c/eng/pipelines/coreclr/templates/perf-job.yml#L171

**Comment by ilonatommy** [comment]
`RuntimeError: Unable to determine the .NET SDK used for net9.0. SDKs found in /mnt/vss/_work/1/performance/CorrelationStaging/payload/dotnet/sdk: ['10.0.100-alpha.1.24468.12']. Major version: 9` [build](https://dev.azure.com/dnceng-public/public/_build/results?buildId=813887&view=logs&j=01945a72-f033-59f1-6eee-baf2cd99ca2f&t=d7a885eb-676a-5c24-61ef-84e00bb2ea51)

**Comment by am11** [comment]
Yup, it's picked up. I have reapplied the workaround (which I removed to see if it's needed after the clone is fixed). Please retry.

**Comment by ilonatommy** [comment]
Worked, performance is green. We should apply it to perf repo and then revert the changes to cloned url.

**Comment by ilonatommy** [comment]
We're stuck on
```
dotnet build /datadisks/disk1/work/A6660968/w/BEFD0A5E/e/performance/src/benchmarks/micro/MicroBenchmarks.csproj --configuration Release --framework net9.0 --no-restore /p:NuGetPackageRoot=/datadisks/disk1/work/A6660968/w/BEFD0A5E/e/performance/artifacts/packages /p:RestorePackagesPath=/datadisks/disk1/work/A6660968/w/BEFD0A5E/e/performance/artifacts/packages /p:UseSharedCompilation=false /p:BuildInParallel=false /m:1 --output /datadisks/disk1/work/A6660968/w/BEFD0A5E/e/performance/artifacts/bin/for-running/MicroBenchmarks /p:BuildingForWasm=true
....
[2024/09/20 12:00:48][WARNING] Benchmark run for framework 'net9.0' contains errors
```
shouldn't we change the arg to be `-f net10.0`? I'm not sure where we're passing it from

**Comment by am11** [comment]
Is it so that before https://github.com/dotnet/runtime/pull/101143, major version change wasn't getting blocked on perf leg? It could be we might have missed considering this detail during that refactoring work. cc @DrewScoggins, @caaavik-msft

One way to move forward could be to disable perf legs and enable it once the major version transition has been completed.

**Comment by lewing** [comment]
> Is it so that before #101143, major version change wasn't getting blocked on perf leg? It could be we might have missed considering this detail during that refactoring work. cc @DrewScoggins, @caaavik-msft
> 
> One way to move forward could be to disable perf legs and enable it once the major version transition has been completed.

Typically, we continue to target 9 but install the net10 sdk during transition periods like this.  I'd rather not disable the perf tests, it has lead to large gaps in testing in the past that included regressions that were extremely difficult to isolate.

**Comment by am11** [comment]
@lewing yup, what I was driving at is, I _think_ before https://github.com/dotnet/runtime/pull/101143,, we were able to do "use net(N) on net(N+1) in perf leg", now it's not possible, or maybe it requires a bit different approach? I haven't looked through last year PRs too closely yet.

**Comment by caaavik-msft** [comment]
> @lewing yup, what I was driving at is, I _think_ before #101143,, we were able to do "use net(N) on net(N+1) in perf leg", now it's not possible, or maybe it requires a bit different approach? I haven't looked through last year PRs too closely yet.

This should still be possible. Looking at the wasm-runtime-perf build logs the only thing that was stopping this from working was the version parsing issue. I think if we just revert the changes to channel_map.py from https://github.com/dotnet/performance/pull/4470 then it should work. If it doesn't, I'd be curious to know what the error is.

**Comment by am11** [comment]
> If it doesn't, I'd be curious to know what the error is.

https://github.com/dotnet/performance/pull/4470#issuecomment-2363599674 <- this was the error before making changes in channel_map. Feel free to push to that PR and just rerun this leg https://dev.azure.com/dnceng-public/public/_build/results?buildId=813887&view=logs&j=59ba1e23-1c3f-5e83-bfa2-2b4c458b95b8&t=606691c3-9bf3-5776-e07f-204fdb23192e (it's already on attempt 7).

**Comment by caaavik-msft** [comment]
After looking into it some more, I think the reason it isn't working is because of the new SDK validation feature (https://github.com/dotnet/BenchmarkDotNet/pull/2523, https://github.com/dotnet/BenchmarkDotNet/pull/2538) that is part of BenchmarkDotNet which will validate that the SDK is installed. The SDK validation feature was introduced 6 months ago and so was not an issue when we went from .NET 8 to 9.

**Comment by lewing** [comment]
> After looking into it some more, I think the reason it isn't working is because of the new SDK validation feature ([dotnet/BenchmarkDotNet#2523](https://github.com/dotnet/BenchmarkDotNet/pull/2523), [dotnet/BenchmarkDotNet#2538](https://github.com/dotnet/BenchmarkDotNet/pull/2538)) that is part of BenchmarkDotNet which will validate that the SDK is installed. The SDK validation feature was introduced 6 months ago and so was not an issue when we went from .NET 8 to 9.

Does this mean the perf runs will fail for all the runtimes but we're only seeing wasm fail because of our validation run?

**Comment by lewing** [comment]
Just a reminder that runtime flow into the sdk is stuck at https://github.com/dotnet/sdk/pull/43061 and this is the next step in unblocking that.

**Comment by caaavik-msft** [comment]
> > After looking into it some more, I think the reason it isn't working is because of the new SDK validation feature ([dotnet/BenchmarkDotNet#2523](https://github.com/dotnet/BenchmarkDotNet/pull/2523), [dotnet/BenchmarkDotNet#2538](https://github.com/dotnet/BenchmarkDotNet/pull/2538)) that is part of BenchmarkDotNet which will validate that the SDK is installed. The SDK validation feature was introduced 6 months ago and so was not an issue when we went from .NET 8 to 9.
> 
> Does this mean the perf runs will fail for all the runtimes but we're only seeing wasm fail because of our validation run?

I ran our full performance test suite, and although it is still running I think it only affects wasm and nativeaot because for them we specify a custom runtime argument ([link to source](https://github.com/dotnet/performance/blob/main/scripts/micro_benchmarks.py#L266-L278)) which specifies which SDK to run against whereas for other runs like coreclr there is no SDK passed in, it just uses the corerun path.

We have a few options here:
1. Make a PR to BDN to add wasmnet100 as a possible runtime. This would need to be synchronised with this PR to change the TFM to net10.0.
2. Make a PR to BDN that allows for disabling the SDK validator
3. It is possible to instead of passing in `--runtimes wasmnet70` to pass in `--runtimes wasm` and it will use the same SDK that is used to run the benchmark project which might work. Unfortunately the BDN codebase has some code that uses this value to determine whether or not to load test-main.js or main.js as that changed in .NET 7. I was thinking a bit of a hacky workaround that might work is to make it so that we rename test-main.js to main.js just before calling BDN and so we wouldn't need to make any BDN changes to get it all working.

I'm going to see if option 3 works first, but if it doesn't then I'll probably proceed with option 2 which might take until Monday to get merged into BDN so we can start using it. One other thing is that it has been quite a while since we last updated our BDN version, I am concerned that if we pull the latest version of BDN that it might cause other things to break and it might take longer than we expected if other bug fixes need to be made. If that happens we might just make a custom build of BDN based off the current version we are using but only with the fix for the SDK validator.

**Comment by am11** [comment]
> Make a PR to BDN that allows for disabling the SDK validator

This will scale better. Let SDK handle the validations and delete those dozen of back and forth mappings. If BDN’s excuse is that public moniker enum used in the attribute, it’s probably worth taking a breaking change and delete the enum with parameterized one `[SupportedMoniker(“net9.0”)`. Hardcoded lists of platform/framework names and versions is not a good design, and we seem to have just too many of them.

**Comment by lewing** [comment]
Agreed, whatever the solution let's assume the same thing will happen again next September and plan for that this time.

**Comment by caaavik-msft** [comment]
Made the BDN PR here https://github.com/dotnet/BenchmarkDotNet/pull/2641

File: eng/pipelines/coreclr/templates/build-perf-bdn-app.yml
Comment (am11) [inline eng/pipelines/coreclr/templates/build-perf-bdn-app.yml:73]:
```suggestion
      git clone https://github.com/dotnet/performance.git --depth 1 -b ${{parameters.perfBranch}} --single-branch
```

**Comment by am11** [comment]
Lets revert the fork pointer, since the branch is merged.

**Comment by lewing** [comment]
:shipit:

**Comment by am11** [comment]
BA is green! 🍏

File: src/libraries/Microsoft.Bcl.Memory/src/Microsoft.Bcl.Memory.csproj
Comment (ericstj) [inline src/libraries/Microsoft.Bcl.Memory/src/Microsoft.Bcl.Memory.csproj:4]: You cannot drop `netstandard2.1` from this package.  That's breaking and the cause of the compatibility error being suppressed.

File: src/libraries/System.Numerics.Tensors/src/CompatibilitySuppressions.xml
Comment (ericstj) [inline src/libraries/System.Numerics.Tensors/src/CompatibilitySuppressions.xml:132]: If you wanted to update to use RC1 baseline this would go away.  That can be done separately.

**Review by ericstj — APPROVED** [review]
If you want to fix the BCL.Memory feedback as a follow up you can, but that is a breaking change that will cause trouble.  It's dropping the facade for .NETStandard2.1 - which means duplicate types when targeting netstandard2.1.

File: src/libraries/System.Numerics.Tensors/src/CompatibilitySuppressions.xml
Comment (lewing) [reply src/libraries/System.Numerics.Tensors/src/CompatibilitySuppressions.xml:132]: We should be now, this was probably done before then.

File: src/libraries/Common/tests/StaticTestGenerator/README.md
Comment (am11) [reply src/libraries/Common/tests/StaticTestGenerator/README.md:25]: Probably incorrect in main as well `net6.0\...\5.0.0`, maybe with last part 10.0.0, it makes a good double-digit example? (.net 6 will be out of support in Nov.)

Comment (ericstj) [reply src/libraries/Common/tests/StaticTestGenerator/README.md:25]: I didn't even notice that other version in the path -- you're right that's wrong as well.  Seems OK to update this just give it a deeper review for correctness.

File: src/libraries/System.Numerics.Tensors/src/CompatibilitySuppressions.xml
Comment (ericstj) [reply src/libraries/System.Numerics.Tensors/src/CompatibilitySuppressions.xml:132]: Referring to updating this version to RC1 : https://github.com/dotnet/runtime/pull/106599/files#diff-9da24614831c308827a1ae533ffea392c97638c261dd42bd0f5226baa136d16eR125
```xml
    <ApiCompatNetCoreAppBaselineVersion>9.0.0-preview.7.24405.7</ApiCompatNetCoreAppBaselineVersion>
```

**Pull request merged**
